### PR TITLE
Upgrade to Angular 19

### PR DESCRIPTION
**Pull request by pawelfras**
- upgrade Angular-related dependencies to v19
- removed 'standalone: true' flag from components, directive and README.md as this is now default value
- update CHANGELOG.md

### CODE REVIEW

File: projects/ngx-infinite-scroll/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## v 18.0.0 (2024/11/20)
```
Comment (SASHAANGELM) [inline projects/ngx-infinite-scroll/CHANGELOG.md:1]: Version mistake, should be `## v 19.0.0 (2024/11/20)`

Comment (pawelfras) [reply projects/ngx-infinite-scroll/CHANGELOG.md:1]: Good catch! Thank you
@orizens Should I also change the lib version in package.json as part of this PR? https://github.com/pawelfras/ngx-infinite-scroll/blob/ceff3d6e6834e9954c9d0b937c2f6f8fa7530b0/projects/ngx-infinite-scroll/package.json#L3

**Review by iamishver — APPROVED** [review]

**Review by SASHAANGELM — APPROVED** [review]

File: projects/ngx-infinite-scroll/src/lib/ngx-infinite-scroll.directive.ts [hunk 2]
```diff
@@ -20,7 +20,6 @@ import {
 
 @Directive({
   selector: '[infiniteScroll], [infinite-scroll], [data-infinite-scroll]',
-  standalone: true
```
Comment (orizens) [inline projects/ngx-infinite-scroll/src/lib/ngx-infinite-scroll.directive.ts:23]: why did you remove this one?

File: yarn.lock
Comment (orizens) [inline yarn.lock:1]: please remove from pr

File: projects/ngx-infinite-scroll/src/lib/ngx-infinite-scroll.directive.ts (hunk 2 above)
Comment (cerealcable) [reply projects/ngx-infinite-scroll/src/lib/ngx-infinite-scroll.directive.ts:23]: With Angular 19, standalone is now default to true.

> By default, Angular components are standalone, meaning that you can directly add them to the imports array of other components. Components created with an earlier version of Angular may instead specify standalone: false in their @Component decorator. For these components, you instead import the NgModule in which the component is defined. See the full [NgModule guide](https://angular.dev/guide/ngmodules) for details.
> 
> Important: In Angular versions before 19.0.0, the standalone option defaults to false.

Because this PR would then require Angular 19, removing this is acceptable since `standalone: true` is the new default.

https://angular.dev/guide/components#imports-in-the-component-decorator

File: yarn.lock
Comment (pawelfras) [reply yarn.lock:1]: Hi @orizens,
Could you help me understand the reason behind removing `yarn.lock` file from this PR?
All previous PRs containing upgrades to new versions of Angular include `yarn.lock` changes.

File: projects/ngx-infinite-scroll/src/lib/ngx-infinite-scroll.directive.ts (hunk 2 above)
Comment (pawelfras) [reply projects/ngx-infinite-scroll/src/lib/ngx-infinite-scroll.directive.ts:23]: Yes, that's the exact reason. Thank you @cerealcable for your answer!

Comment (orizens) [reply projects/ngx-infinite-scroll/src/lib/ngx-infinite-scroll.directive.ts:23]: we're not using yarn for dev.

File: yarn.lock
Comment (orizens) [reply yarn.lock:1]: we're not using yarn for dev.

**Comment by orizens** [comment]
@pawelfras  thank you 
v19 published
